### DISCUSSION TITLE
Blinky example on ch32x035f8u6

### DISCUSSION THREAD

**Discussion by anti-social** [ch32-rs/ch32-hal#35] [General]
I'm trying to run a `blinky` example on my `ch32x035f8u6` board from [WeActStudio](https://github.com/WeActStudio/WeActStudio.CH32X035CoreBoard). But the LED is constantly shines. If I exchange `set_low` and `set_high` in the example's code it does not shine at all. I tried to do blocking delays and it works as expected. So I wonder how can I debug what's going wrong?

**Comment by romainreignier** [comment]
Might be related to the issue I had [here](https://github.com/ch32-rs/ch32-hal/issues/6#issuecomment-2175649987).
Try to replace the `embassy-time` feature `tick-hz-1_000_000` with `tick-hz-10_000` in the `ch32-hal` [`Cargo.toml`](https://github.com/ch32-rs/ch32-hal/blob/5df0e89a399780a6d0476535b12bc3180ed49703/Cargo.toml#L43).

**Reply by anti-social** [reply]
Yep, it works, thank you!

As I understand it is not possible to change a feature of a transitive dependency so I need to have a fork of `ch32-hal`?

**Reply by anti-social** [reply]
By the way even with `tick-hz-10_000` it stuck after a day of work

**Reply by romainreignier** [reply]
Interesting, with higher frequencies, I had issues after some seconds.
You may open an issue about this, I think we have an overflow issue somewhere in the embassy-time timer code.

**Comment by romainreignier** [comment]
For now, yes. I think this is something @andelf should think about.

**Reply by andelf** [reply]
Thanks the the report. I'll fix this.
